### PR TITLE
Sanitation of the empty database

### PR DESCRIPTION
**Pull request by Simply007**
When setting up my fork of the site I got an exception because code depends that data from the database returns at least empty object, but when the db is empty, `val()` method returns null, that cause to exception when calling `Object.values(null)`.

Also added cross-env package, since I am working on windows, setting process environment is done differently.
crosss-env package is unifying the setup of unix like and window like systems.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -16,10 +16,12 @@
     }
   },
   "engines": {
-    "node": "8.11.4",
-    "yarn": "1.9.4"
+    "node": "^8.11.1",
```
Comment (flxwu) [inline package.json]: please leave the engine specification as-is, you can change it in your fork :) But we need it for the installs on heroku.

**Review by flxwu — APPROVED** [review]
thanks! please revert your change to the engine specs, I'll be happy to merge this then!

**Review by flxwu — CHANGES_REQUESTED** [review]
node engine specs

File: package.json (hunk 1 above)
Comment (Simply007) [reply package.json]: OK

**Comment by Simply007** [comment]
I am sorry. I have forgotten, that I have used my master branch for this pull request - now it should be all fixed.

**Comment by flxwu** [comment]
Please remove the firebase dependencies that you added as I don't see that they're used anywhere. 
Since you then didn't add any new package, could you please discard the yarn lockfile using `git checkout upstream/master -- yarn.lock` ? 🙂 (assuming that upstream is this repository, if not add it with `git remote add upstream https://github.com/cssfrankfurt/hacktoberfest2018.git`)

thank you! 😊

File: package.json [hunk 2]
```diff
@@ -20,6 +20,8 @@
     "yarn": "1.9.4"
   },
   "dependencies": {
+    "@firebase/app-types": "^0.3.2",
+    "@firebase/database-types": "^0.3.2",
```
Comment (flxwu) [inline package.json]: why did you add those? I don't see that they're used anywhere... 😅

**Review by flxwu — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (Simply007) [reply package.json]: They are used basically for intellisense in IDE. 
* Probably they should be placed in dev dependencies. I didn't want them installed in global.

**Comment by flxwu** [comment]
awesome! thanks! @Simply007
